### PR TITLE
fix(less): type definitions on options and rootFileInfo

### PR DESCRIPTION
**Pull request by chigix**
Signed-off-by: Richard Lea <chigix@zoho.com>

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://lesscss.org/usage/#less-options
https://github.com/less/less.js/blob/master/lib/less/import-manager.js#L10
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/less/index.d.ts [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+        /** path to append to normal URLs for this node */
         rootpath: string;
+        /** path to the current file, absolute */
         currentDirectory: string;
+        /** absolute path to the entry file */
         entryPath: string;
+        /** filename of the base file */
         rootFilename: string;
+        /** whether the file should not be output and only output parts that are referenced */
+        reference: boolean;
```
Comment (dwaxweiler) [inline types/less/index.d.ts]: Where is this attribute mentioned?

**Comment by dwaxweiler** [comment]
Wow, you have added most options! :+1: Could you add a few small tests please?

**Review by dwaxweiler — COMMENTED** [review]
Wow, you have added most options! :+1: Could you add a few small tests please?

File: types/less/index.d.ts (hunk 1 above)
Comment (chigix) [reply types/less/index.d.ts]: Mentioned in the document comment in official repository: 
https://github.com/less/less.js/blob/master/lib/less/import-manager.js#L17

**Comment by chigix** [comment]
@dwaxweiler Actually, I can not understand very well about the style of the testing file. I can not see what it is testing or any assertion to check result value.

File: types/less/index.d.ts [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 // Project: http://lesscss.org/
 // Definitions by: Tom Hasner <https://github.com/thasner>
 //                 Pranay Prakash <https://github.com/pranaygp>
 //                 Daniel Waxweiler <https://github.com/dwaxweiler>
+//                 Richard Lea <https://github.com/chigix>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace Less {
+    // https://github.com/less/less.js/blob/master/lib/less/import-manager.js#L10
     interface RootFileInfo {
```
Comment (dwaxweiler) [inline types/less/index.d.ts]: Where do we use this interface?

File: types/less/index.d.ts (hunk 1 above)
Comment (dwaxweiler) [reply types/less/index.d.ts]: Thanks!

**Review by dwaxweiler — CHANGES_REQUESTED** [review]
@chigix I have added another question. Indeed, the test file looks a bit messy. You could add the options to the following expression: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/049be10668f96773869275d26901004f5158e359/types/less/less-tests.ts#L33

File: types/less/index.d.ts (hunk 2 above)
Comment (dwaxweiler) [reply types/less/index.d.ts]: True, I have overlooked this one!

**Review by dwaxweiler — APPROVED** [review]
@chigix Thanks for expanding the tests! :+1:

**Pull request merged**